sim/core/sim.go

```diff
@@ -220,10 +220,28 @@ func (sim *Simulation) RunOnce() {
 			// This path is not currently used by individual shaman sim.
 			if pa.NextActionAt == NeverExpires {
 				sim.pendingActions = sim.pendingActions[1:] // cut off front
+			} else {
+				handled := false
+				for i, v := range sim.pendingActions {
+					if i == 0 {
+						continue
+					}
+					if v.NextActionAt >= pa.NextActionAt {
+						handled = true
+						if i == 1 {
+							sim.pendingActions[0] = pa
+							break // just leave it there
+						}
+						copy(sim.pendingActions, sim.pendingActions[1:i])
```

Comment:
I think this forgets to copy over any pending actions after i.

Reply:
doesnt need to - you are inserting into the list at i -- this keeps the slice sorted with the lowest NextActionAt always at element 0.

Reply:
Ah yes I see, I misunderstood. This LGTM.

When we have a raid sim, using binary search might be better here. On the other hand, maybe looping is better even in that case because the # of pending actions will still be not that large and loops are usually better-optimized.